docs/1_how_it_works/1_1_how_it_works.md

```diff
@@ -0,0 +1,50 @@
+# How Finetuner Works
+
+## Contrastive Metric Learning 👓
+
+From an algorithmic perspective,
+**Finetuner** leverages a contrastive metric learning approach to improve your model.
+How does it work?
+
+### Step 1: Convert your model into an Embedding Model
+
+Finetuner interprets your model architecture,
+removes the default *head*, applies *pooling* and freezes layers that do not need to be trained..
+For example, if your pre-trained model is trained for cat-dog classification,
+Finetuner is going to remove the cat-dog classifier and turn your model into an *embedding model*.
+
+This embedding model does not make predictions or outputs a probability,
+but instead outputs a feature vector to represent your data
+
+### Step 2: Triplets Construction and Training on-the-fly
+
+Finetuner "looks" into the label of your training data.
+Each `Document` and `Document`s share the same ****`finetuner_label` is considered as a *Positive*.
+In the meanwhile, each `Document` and `Documents` which share a different `finetuner_label` is considered as a *Negative*.
+
+During model fine-tuning, Finetuner is creating *Triplets*  ``(anchor, positive, negative)`` on-the-fly.
+Finetuner then uses the triplets to perform training,
```

Comment:
I think this part needs a bit more clarity.

Some suggestions:

Finetuner works on labeled data. It expects a `DocumentArray` consisting of `Documents` where each one contains `finetuner_label` corresponding to a class of a specific training example.

During the fine-tuning, Finetuner creates *Triplets*  ``(anchor, positive, negative)`` on-the-fly.
For each *anchor*, which can be any training example, Finetuner looks for a `Document` with the same `finetuner_label` (*positive*), and a `Document` with a different `finetuner_label` (*negative*).